python_modules/dagster/dagster/core/storage/config.py

```diff
@@ -0,0 +1,35 @@
+from dagster.config import Field, Permissive, Selector
+from dagster.config.source import IntSource, StringSource
+
+
+def mysql_config():
+    return Selector(
+        {
+            "mysql_url": StringSource,
+            "mysql_db": {
+                "username": StringSource,
+                "password": StringSource,
+                "hostname": StringSource,
+                "db_name": StringSource,
+                "port": Field(IntSource, is_required=False, default_value=3306),
+            },
+        }
+    )
+
+
+def pg_config():
+    return {
+        "postgres_url": Field(StringSource, is_required=False),
+        "postgres_db": Field(
+            {
+                "username": StringSource,
+                "password": StringSource,
+                "hostname": StringSource,
+                "db_name": StringSource,
+                "port": Field(IntSource, is_required=False, default_value=5432),
+                "params": Field(Permissive(), is_required=False, default_value={}),
+            },
+            is_required=False,
+        ),
+        "should_autocreate_tables": Field(bool, is_required=False, default_value=True),
+    }
```

Comment:
comment something about keeping these in sync with their equivalents in the libraries? 

related there should be a test to keep them in sync as well (or just have the libs source from core?)

Reply:
I actually refactored these so it's using the same config in both places (`dagster-postgres` lib now reads from this field).